plugins/node/opentelemetry-instrumentation-winston/package.json

```diff
@@ -58,12 +59,14 @@
     "sinon": "15.2.0",
     "test-all-versions": "6.0.0",
     "ts-mocha": "10.0.0",
-    "typescript": "4.4.4",
-    "winston": "3.3.3",
-    "winston2": "npm:winston@2.4.7"
+    "typescript": "4.4.4"
   },
   "dependencies": {
-    "@opentelemetry/instrumentation": "^0.46.0"
+    "@opentelemetry/api-logs": "^0.45.1",
+    "@opentelemetry/core": "1.18.1",
+    "@opentelemetry/instrumentation": "^0.46.0",
+    "winston": "3.3.3",
```

Comment:
Most (all?) the winston@3 transports I see have a dependency just on `winston-transport`. Could this be changed to only depend on it, and not take a runtime dependency on `winston` itself?

Reply:
The guild [here](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/GUIDELINES.md#instrumented-package-dependency) explains why instrumentations should not depend on the instrumented library.

I think the instrumentation has to find the winston-transport instances (there might be several of different versions) via the hooking mechanism used for patching. If this isn't possible some other mechanism should be used to avoid depending on the instrumented library.

Reply:
@Flarna If this PR changes to dep on `winston-transport` would that satisfy the guidelines? This is instrumenting `winston`, not `winston-transport`.

One possible use-case of this package -- as described at https://github.com/open-telemetry/opentelemetry-js-contrib/blob/4c6e81347bb1dbc0c87a0de484a48354cef46cb2/plugins/node/opentelemetry-instrumentation-winston/README.md#using-opentelemetrywinstontransport-without-instrumentation -- will be for a user to directly use the `OpenTelemetryTransportv3` class to configure their Winston logger. In that case there isn't necessary an active Winston instrumentation. In other words, I don't think it is possible to rely on being able to get `winston-transport` via the hooking mechanism.

Reply:
As long as `@opentelemetry/instrumentation-winston`  is part of `@opentelemetry/auto-instrumentations-node` I would avoid such dependencies because
* it adds winston (incl deps) even in projects where winston is not used
* it has the risk of causing double installs because version ranges do not match
* might cause issuse with types (see #932)

Maybe we should remove winston instrumentation from `@opentelemetry/auto-instrumentations-node`. OptIn will be likely done by winston users only.
Or split packages into the "classic" instrumentation (part of auto package) and the transport extension (opt-in).